scripts/ci/run_ci.sh

```diff
@@ -54,22 +66,90 @@ while getopts ":pm:b:r:M:" opt; do
 done
 
 DOC_MATRIX=${MATRIX_BUILDS}
-if [ -z "$BRANCH" ]; then
+if [ -n "$MAIN_CI" -a -z "$BRANCH" ]; then
```

Comment:
It feels like wrapping these 3 ifs in a `if [ -n  "$MATRIX_CI" ]` instead of repeating it 3 times would have been cleaner/more obvious to the reader.